opentok-whiteboard.js

```diff
@@ -96,90 +93,95 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
             };
             
             scope.erase = function () {
-                //scope.color = element.css("background-color") || "#fff";
-                scope.lineWidth = 50;
                 scope.erasing = true;
             };
             
             scope.capture = function () {
                 if (iOS) {
                     // On iOS you can put HTML in a mailto: link
-                    window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
+                    $window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
                 } else {
                     // We just open the image in a new window
-                    window.open(canvas.toDataURL('image/png'));
+                    $window.open(canvas.toDataURL('image/png'));
                 }
             };
 
             scope.undo = function () {
                 if (!undoStack.length)
                     return;
-                var undodata = undoStack.pop();
-                undoWhiteBoard(undodata);
-                redoStack.push(undodata);
-                sendUpdate('otWhiteboard_undo', undodata);
+                var data = undoStack[undoStack.length - 1];
+                undoWhiteBoard(data);
+                sendUpdate('otWhiteboard_undo', data);
             };
 
             var undoWhiteBoard = function (data) {
```

Comment:
Actually, why not leave the `var undodata = undoStack.pop();` line here and then pass that data in and you don't need to do it in the undoWhiteBoard method.